lib/bugsnag/endpoint_validator.rb

```diff
@@ -0,0 +1,80 @@
+module Bugsnag
+  # @api private
+  class EndpointValidator
+    def self.validate(endpoints)
+      # ensure we have an EndpointConfiguration object
+      return Result.missing_urls unless endpoints.is_a?(EndpointConfiguration)
+
+      # check for missing URLs
+      return Result.missing_urls if endpoints.notify.nil? && endpoints.sessions.nil?
+      return Result.missing_notify if endpoints.notify.nil?
+      return Result.missing_session if endpoints.sessions.nil?
+
+      # check for empty URLs
+      return Result.invalid_urls if endpoints.notify.empty? && endpoints.sessions.empty?
+      return Result.invalid_notify if endpoints.notify.empty?
+      return Result.invalid_session if endpoints.sessions.empty?
+
+      Result.valid
+    end
+
+    # @api private
+    class Result
+      # rubocop:disable Layout/LineLength
+      MISSING_URLS = "Invalid configuration. endpoints must be set with both a notify and session URL. Bugsnag will not send any requests.".freeze
+      MISSING_NOTIFY_URL = "Invalid configuration. endpoints.sessions cannot be set without also setting endpoints.notify. Bugsnag will not send any requests.".freeze
+      MISSING_SESSION_URL = "Invalid configuration. endpoints.notify cannot be set without also setting endpoints.sessions. Sessions will not be sent to Bugsnag.".freeze
```

Comment:
I think this mixes the target of who "Bugsnag" is - i.e. the notifier or the endpoint - compared to the other messages.

```suggestion
      MISSING_SESSION_URL = "Invalid configuration. endpoints.notify cannot be set without also setting endpoints.sessions. Bugsnag will not send any sessions.".freeze
```